include/fmt/core.h

```diff
@@ -933,7 +959,18 @@ template <typename T> class buffer {
   }
 
   /** Appends data to the end of the buffer. */
-  template <typename U> void append(const U* begin, const U* end);
+  template <typename U>
+  FMT_CONSTEXPR void append(const U* begin, const U* end) {
+    while (begin != end) {
+      auto count = to_unsigned(end - begin);
+      try_reserve(size_ + count);
+      auto free_cap = capacity_ - size_;
+      if (free_cap < count) count = free_cap;
+      auto out = make_checked(ptr_ + size_, count);
+      for (size_t i = 0; i < count; ++i) *out++ = *begin++;
```

Comment:
This used to be `std::uninitialized_copy_n(begin, count, make_checked(ptr_ + size_, count));`, but `uninitialized_copy_n` is not constexpr, so this is changed to use a plan for loop. Please let me know if this has any issues.

Reply:
This is likely a regression. Please revert it and other changes to core and limit the changes to the compile-only API.

Reply:
Since you are likely to have C++20, you can try to use https://en.cppreference.com/w/cpp/types/is_constant_evaluated

Also
```
#if defined(__cpp_lib_constexpr_string) &&   \
    __cpp_lib_constexpr_string >= 201907L && \
    defined(__cpp_lib_constexpr_iterator) && \
+    __cpp_lib_constexpr_iterator >= 201811L && \
+ defined(__cpp_lib_is_constant_evaluated) && \
+ __cpp_lib_is_constant_evaluated >= 201811L
```